qiita_db/study.py

```diff
@@ -825,39 +831,58 @@ def ebi_submission_status(self):
         -------
         str
             The study EBI submission status
-        """
-        with qdb.sql_connection.TRN:
-            sql = """SELECT ebi_submission_status
-                     FROM qiita.{0}
-                     WHERE study_id = %s""".format(self._table)
-            qdb.sql_connection.TRN.add(sql, [self.id])
-            return qdb.sql_connection.TRN.execute_fetchlast()
 
-    @ebi_submission_status.setter
-    def ebi_submission_status(self, value):
-        """Sets the study's EBI submission status
-
-        Parameters
-        ----------
-        value : str {%s}
-            The new EBI submission status
-
-        Raises
-        ------
-        ValueError
-            If the status is not known
+        Notes
+        -----
+        There are 4 possible states: 'not submitted', 'submitting',
+        'submitted' & 'failed'. We are going to assume 'not submitted' if the
+        study doesn't have an accession, 'submitted' if it has an accession,
+        'submitting' if there are submit_to_EBI jobs running using the study
+        artifacts, & 'failed' if there are artifacts with failed jobs without
+        successful ones.
         """
-        if not (value in _VALID_EBI_STATUS or
-                value.startswith('failed')):
-            raise ValueError("Unknown status: %s" % value)
+        status = 'not submitted'
         with qdb.sql_connection.TRN:
-            sql = """UPDATE qiita.{}
-                     SET ebi_submission_status = %s
-                     WHERE study_id = %s""".format(self._table)
-            qdb.sql_connection.TRN.add(sql, [value, self.id])
-            qdb.sql_connection.TRN.execute()
-
-    ebi_submission_status.__doc__.format(', '.join(_VALID_EBI_STATUS))
+            if self.ebi_study_accession:
+                status = 'submitted'
+
+            plugin = qdb.software.Software.from_name_and_version(
+                'Qiita', 'alpha')
+            cmd = plugin.get_command('submit_to_EBI')
+
+            sql = """SELECT processing_job_id, command_parameters->>'artifact',
+                        processing_job_status
+                     FROM qiita.processing_job
+                     LEFT JOIN qiita.processing_job_status
+                        USING (processing_job_status_id)
+                     WHERE command_parameters->>'artifact' IN (
+                        SELECT artifact_id::text
+                        FROM qiita.study_artifact
+                        WHERE study_id = {0}) AND command_id = {1}""".format(
+                            self._id, cmd.id)
+            qdb.sql_connection.TRN.add(sql)
+            jobs = defaultdict(dict)
+            for info in qdb.sql_connection.TRN.execute_fetchindex():
+                jid, aid, js = info
+                jobs[js][aid] = jid
+
+            if 'queued' in jobs or 'running' in jobs:
```

Comment:
Instead of putting the logic in python, it might be possible to use postgres to only search for the jobs that have a status of "error", and if the query is empty then you can assume that it's still submitting, as there's not an accession available.

Maybe it's more complicated, just thought I would share this idea.

Reply:
I think the only issue with this option is that we will still have to deal with those jobs in a success state.

Reply:
Ah, got it, thanks!